src/store/actions.ts

```diff
@@ -1053,31 +1051,33 @@ export const fetchMRASProfile = async ({ commit, getters }): Promise<any> => {
 /** Submits an edited NR or a new name submission. */
 export const submit: any = async ({ commit, getters }): Promise<any> => {
   // TODO: Handle Edit Submit
-  // if (getters.isEditMode) {
-  //   if (await newRequestModule.patchNameRequests()) {
-  //     if (await newRequestModule.checkinNameRequest()) {
-  //       newRequestModule.mutateDisplayedComponent('Success')
-  //       await sleep(1000) // wait for a second to show the update success
-  //       await this.fetchNr(+getters.getNrId)
-  //     }
-  //   }
-  // } else {
-  let request
-  if (!getters.getNrId) {
-    request = await postNameRequests({ commit, getters }, 'draft')
+  if (getters.isEditMode) {
+    // TODO-CAM: Refactor the way these async requests are used to provide conditional booleans
+    // @ts-ignore
+    if (await patchNameRequests()) {
+      // @ts-ignore
+      if (await checkinNameRequest()) {
+        commit('mutateDisplayedComponent', 'Success')
+        await sleep(1000) // wait for a second to show the update success
+        await this.fetchNr(+getters.getNrId)
+      }
+    }
   } else {
-    if (!getters.isEditMode && [NrState.COND_RESERVED, NrState.RESERVED].includes(getters.getNrState)) {
-      request = await getNameRequest(getters.getNrId)
-      if (request?.stateCd === NrState.CANCELLED) {
-        await setActiveComponent('Timeout')
-        return
+    let request
+    if (!getters.getNrId) {
+      request = await postNameRequests({ commit, getters }, 'draft')
+    } else {
+      if (!getters.isEditMode && [NrState.COND_RESERVED, NrState.RESERVED].includes(getters.getNrState)) {
+        request = await getNameRequest(getters.getNrId)
+        if (request?.stateCd === NrState.CANCELLED) {
+          await setActiveComponent('Timeout')
+          return
+        }
       }
+      request = await putNameReservation(getters.getNrId)
     }
-    request = await putNameReservation(getters.getNrId)
+    if (request) await paymentModule.togglePaymentModal(true)
```

Comment:
Is error handling (if the PUT failed) in putNameReservation() ?

Reply:
Yes, turns out it is!

Reply:
General soapboxing: This is not a great architecture because, after all this time in the code, we still don't know how/where/when errors are handled. A better architecture would have small, concise network access methods/services and let the "logic code" (middle-tier, if UI code is upper tier and network code is lower tier) handle, well, the logic, including error handling.

There is value in having the network functions report their own errors -- as long as it's done consistently (which it isn't). But I still think low-level functions should not have middle- or high-level logic.

This can be something we clean up... later. Or as we go, if something's really bugging us. I dunno.

End rant.

Reply:
😆 I think the rants are the only thing keeping us sane. 
Please, rant away!

And yes to ALL of the above.